build.sbt

```diff
@@ -318,10 +318,18 @@ lazy val commonSettings = ReleasePlugin.extraReleaseCommands ++ Seq(
         <url>http://github.com/fwbrasil/</url>
       </developer>
     </developers>)
-) ++ update2_12
+) ++ `scala 2.12 hacks`
 
-lazy val update2_12 = Seq(
-  scalacOptions -= (
-    if (scalaVersion.value.startsWith("2.12.")) "-Xfatal-warnings" else ""
-  )
+lazy val `scala 2.12 hacks` = Seq(
+  scalacOptions := {
+    val orig = scalacOptions.value
+    CrossVersion.partialVersion(scalaVersion.value) match {
+      case Some((2, 12)) => Seq("-Xsource:2.11") ++ orig.map {
```

Comment:
We have a problem with that. I have put this flag only to core module and it didn't work. Imho, forcing users to add this flag into their builds doesn't make sense.